### ISSUE TITLE
Explorer throws CORS error when switching to different API server other than HIRO's

### ISSUE THREAD

**Issue by ghost** [hirosystems/explorer#848]
**Describe the bug**
When starting the explorer with NEXT_PUBLIC_MAINNET_API_SERVER environment variable pointing to a non-hiro API node it throws the following CORS error:
```
Access to fetch at 'https://stacks-api.burnchain.io/extended/v1/tx/mempool?limit=10&offset=0&unanchored=true' from origin 'http://localhost:3000' has been blocked by CORS policy: Request header field x-hiro-product is not allowed by Access-Control-Allow-Headers in preflight response.
fetch.ts?b411:5          GET https://stacks-api.burnchain.io/extended/v1/tx/mempool?limit=10&offset=0&unanchored=true net::ERR_FAILED
```

A similar error is thrown when attempting to add a 3rd party API node to https://explorer.stacks.co
```
Access to fetch at 'https://stacks-api.burnchain.io/v2/info' from origin 'https://explorer.stacks.co' has been blocked by CORS policy: Request header field x-hiro-product is not allowed by Access-Control-Allow-Headers in preflight response.
_app-8eb23ca60de3fec7.js:1          GET https://stacks-api.burnchain.io/v2/info net::ERR_FAILED
```

**To Reproduce**
1. Go to 'https://explorer.stacks.co'
2. Click on 'Network -> Add Network'
3. Enter following values 'Name: XYZ, URL: https://stacks-api.burnchain.io'

**Expected behavior**
The alternate API node works with the Explorer.

**Screenshots**
![image](https://user-images.githubusercontent.com/5003742/188253391-df4bee2f-667a-4d8a-9d71-bec0b84245c3.png)
![image](https://user-images.githubusercontent.com/5003742/188253400-a6ff66e6-6591-466e-a5a6-3b5f548233ae.png)

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Chrome
 - Version 104.0.5112.101 (Official Build) (x86_64)

**Smartphone (please complete the following information):**
**Additional context**

**Comment by andresgalante** [comment]
I think this is fixed, please re open the issue if it isn't

**Comment by muthusleuths** [comment]
Not sure if this needs to be re-opened or a new issue is needed. It seems this is happening again:

<img width="1652" alt="Screen Shot 2024-08-31 at 2 22 52 AM" src="https://github.com/user-attachments/assets/e8871036-14a5-43b1-91e9-643edaf9b601">
<img width="1659" alt="Screen Shot 2024-08-31 at 2 23 04 AM" src="https://github.com/user-attachments/assets/6e5422e9-7e39-4348-8fa5-d4c8b0e01ae6">
<img width="1654" alt="Screen Shot 2024-08-31 at 2 23 25 AM" src="https://github.com/user-attachments/assets/0abc8363-e8d8-4831-80ea-f8ab1cba437a">
